### PR TITLE
fix client to use correct port for HTTP/HTTPS

### PR DESCRIPTION
**Pull request by takeon8**
@rajatarya @rbkreisberg

### CODE REVIEW

**Comment by takeon8** [comment]
the rest of the client repos are fine

File: index.js [hunk 1]
```diff
@@ -22,6 +22,10 @@ var PredictiveServiceClient = function(end_point, api_key, should_verify_certifi
   this.end_point = protocol_uri["uri"];
   this.protocol = protocol_uri["protocol"];
   this.timeout = 10000; // default to 10 seconds timeout
+  this.port = 80; // default port 80
```
Comment (rbkreisberg) [inline index.js]: there's duplicated logic here.

remove line 21, 22, 23, 25, 26, 27, 28

replace them with:
```
this.setEndpoint(end_point);
```

If you want to initialize those instance variables at the top of the constructor, that's fine.

File: index.js [hunk 2]
```diff
@@ -38,8 +42,10 @@ PredictiveServiceClient.prototype.setEndpoint = function(end_point) {
   var protocol_uri = PredictiveServiceClient._getProtocol(end_point);
```
Comment (rbkreisberg) [inline index.js]: not critical: this is including a utility function (with no dependency on the object) in an object.  Consider moving such functions out of the object and having static function declarations at the top of the file.  

``` 
// definition at top of script
function getProtocolURI(endpoint) {
...
}

... inside object
this.endpoint = getProtocolURI(endpoint);

File: index.js (hunk 1 above)
Comment (takeon8) [inline index.js]: done

**Comment by takeon8** [comment]
@rbkreisberg

**Comment by rbkreisberg** [comment]
@takeon8 great job.  this code looks really easy to test and use.

merge it.

**Pull request merged**